analyzer/src/main/kotlin/managers/Yarn2.kt

```diff
@@ -63,20 +93,56 @@ class Yarn2(
      */
     private val yarn2ExecutablesByPath: MutableMap<File, String> = mutableMapOf()
 
-    /**
-     * A NPM package manager, used to call NPM when Yarn 2+ does not provide enough information.
-     */
-    private val delegateNpmAnalyzer = Npm("npm", analysisRoot, analyzerConfig, repoConfig)
+    private val disableRegistryCertificateVerification = analyzerConfig
+        .getPackageManagerConfiguration(managerName)?.options
+        ?.get(OPTION_DISABLE_REGISTRY_CERTIFICATE_VERIFICATION).toBoolean()
 
     companion object {
         // The name of Yarn 2+ resource file.
         const val YARN2_RESOURCE_FILE = ".yarnrc.yml"
 
         // The property in `.yarnrc.yml`containing the path to the Yarn2+ executable.
         const val YARN_PATH_PROPERTY_NAME = "yarnPath"
+
+        // The name of the scope with the regular dependencies.
+        private enum class DependencyType(val type: String) {
+            DEPENDENCIES("dependencies"),
+            DEV_DEPENDENCIES("devDependencies")
+        }
+
+        // The pattern to extract rawName, type and version from a Yarn 2+ locator e.g. @babel/preset-env@npm:7.11.0.
+        private val EXTRACT_FROMLOCATOR_PATTERN = Regex("(.+)@(\\w+):(.+)")
+
+        // The pattern to match Artifactory API path.
+        private val ARTIFACTORY_API_PATH_PATTERN = Regex("(.*artifactory.*)/api/npm/(.*)")
+
+        private const val BULK_DETAILS_SIZE = 1000
     }
 
-    override fun hasLockFile(projectDir: File) = hasYarnLockFile(projectDir)
+    /**
+     * A builder to build the dependency graph of the project.
+     */
```

Comment:
These could all use single-line `//` comments as they are not part of the public API anyway.